docs/hoon/library/1a.md

```diff
@@ -64,7 +64,469 @@ Source
 
     ++  dec                                                 ::  decrement
       ~/  %dec
-      |=  a/@
+      |=  a=@
+      ~_  leaf+"decrement-underflow"
+      ?<  =(0 a)
+      =+  b=0
+      |-  ^-  @
+      ?:  =(a +(b))  b
+      $(b +(b))
+
+Examples
+--------
+
+    > (dec 7)
+    6
+    > (dec 0)
+    ! decrement-underflow
+    ! exit
+
+
+
+***
+### `++div`
+
+Divide
+
+Computes `a` divided by `b`.
+
+Accepts
+-------
+
+`a` is an atom.
+
+`b` is an atom.
+
+Produces
+--------
+
+An atom.
+
+Source
+------
+
+    ++  div                                                 ::  divide
+      ~/  %div
+      |:  [a=`@`1 b=`@`1]
+      :>  quotient
+      ^-  @
+      ~_  leaf+"divide-by-zero"
+      ?<  =(0 b)
+      =+  c=0
+      |-
+      ?:  (lth a b)  c
+      $(a (sub a b), c +(c))
+    ::
+
+Examples
+--------
+
+    > (div 4 2)
+    2
+    > (div 17 8)
+    2
+    > (div 20 30)
+    0
+
+
+***
+### `++gte`
+
+Greater-than/equal
+
+Tests whether `a` is greater than a number `b`.
+
+Accepts
+-------
+
+`a` is an atom.
+
+`b` is an atom.
+
+Produces
+--------
+
+A boolean.
+
+Source
```

Comment:
We shouldn't duplicate the source; instead, prepend documentation to the actual source.

Reply:
Oh. I had a moment of delirium and for some reason thought this was in hoon.hoon. I think we do want to eventually move to a literate programming style where these docs would be generated from comments in hoon.hoon itself, but I guess that isn't happening yet.

My apologies for spouting irrelevant noise at you guys.

Reply:
Don't be too hard on yourself, this is definitely relevant noise!